src/main/g8/src/main/java/com/lightbend/akka/http/sample/UserRoutes.java

```diff
@@ -45,66 +45,79 @@ public UserRoutes(ActorSystem system, ActorRef userRegistryActor) {
      * This method creates one route (of possibly many more that will be part of your Web App)
      */
     //#all-routes
-    //#users-get-post
-    //#users-get-delete
     public Route routes() {
         return route(pathPrefix("users", () ->
-                //#users-get-delete
-                route(
-                        pathEnd(() ->
-                                route(
-                                        get(() -> {
-                                                    Future<UserRegistryActor.Users> futureUsers = Patterns.ask(userRegistryActor, new UserRegistryMessages.GetUsers(), timeout)
-                                                            .mapTo(classTag(UserRegistryActor.Users.class));
-                                                    return onSuccess(() -> toJava(futureUsers),
-                                                            users -> complete(StatusCodes.OK, users, Jackson.marshaller()));
-                                                }
-                                        ),
-                                        post(() ->
-                                                entity(Jackson.unmarshaller(User.class), user -> {
-                                                    Future<ActionPerformed> userCreated =
-                                                            Patterns.ask(userRegistryActor, new CreateUser(user), timeout)
-                                                                    .mapTo(classTag(ActionPerformed.class));
-                                                    return onSuccess(() -> toJava(userCreated),
-                                                            performed -> {
-                                                                log.info("Created user [{}]: {}", user.getName(), performed.getDescription());
-                                                                return complete(StatusCodes.CREATED, performed, Jackson.marshaller());
-                                                            });
-                                                }))
-                                )
-                        ),
-                        //#users-get-post
-                        //#users-get-delete
-                        path(PathMatchers.segment(),
-                                name -> route(
-                                                get(() -> {
-                                                    //#retrieve-user-info
-                                                    Future<Optional> maybeUser = Patterns.ask(userRegistryActor, new UserRegistryMessages.GetUser(name), timeout)
-                                                            .mapTo(classTag(Optional.class));
+            route(
+                usersGetPost(),
+                usersGetDelete()
```

Comment:
usersGetDelete is a bit of a confusing name, perhaps match the name here and separate out 2 helper functions for the 'get' and 'delete' cases?

Reply:
Sure! I wasn't very inspired and just used the names of the paradox's tags. :-)

Reply:
Hmmm, I'm not really sure what names to use. I think this could be taken one step further as each of the methods `usersGetPost` and `usersGetDelete` actually has two routes. Extracting to a total of 4 separate methods (so there is one method per action) will make it a bit too granular and hard to setup for paradox to print.

I think I'll go with: `userCollectionRoutes` vs `singleUserRoutes` and add some javadoc on the methods.